src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,341 @@
+            CancellationToken cancellationToken)
+        {
+            return GetDocuments(options, out duration, cancellationToken);
+        }
+
+        private Dictionary<Slice, List<(Slice lowerId, string Id)>> GetDocuments(ArchivedDocumentsOptions options, out Stopwatch duration,
+            CancellationToken cancellationToken)
+        {
+            var count = 0;
+            var currentTicks = options.CurrentTime.Ticks;
+
+            var archiveTree = options.Context.Transaction.InnerTransaction.ReadTree(DocumentsByArchive);
+            using (var it = archiveTree.Iterate(false))
+            {
+                if (it.Seek(Slices.BeforeAllKeys) == false)
+                {
+                    duration = null;
+                    return null;
+                }
+
+                var toArchive = new Dictionary<Slice, List<(Slice LowerId, string Id)>>();
+                duration = Stopwatch.StartNew();
+
+                do
+                {
+                    var entryTicks = it.CurrentKey.CreateReader().ReadBigEndianInt64();
+                    if (entryTicks > currentTicks)
+                        break;
+
+                    var ticksAsSlice = it.CurrentKey.Clone(options.Context.Transaction.InnerTransaction.Allocator);
+
+                    var docsToArchive = new List<(Slice LowerId, string Id)>();
+
+                    using (var multiIt = archiveTree.MultiRead(it.CurrentKey))
+                    {
+                        if (multiIt.Seek(Slices.BeforeAllKeys))
+                        {
+                            do
+                            {
+                                if (cancellationToken.IsCancellationRequested)
+                                    return toArchive;
+
+                                var clonedId = multiIt.CurrentKey.Clone(options.Context.Transaction.InnerTransaction.Allocator);
+
+                                using (var document = _database.DocumentsStorage.Get(options.Context, clonedId,
+                                           DocumentFields.Id | DocumentFields.Data | DocumentFields.ChangeVector))
+                                {
+                                    if (document is null || document.TryGetMetadata(out var metadata) == false || HasPassed(metadata, options.CurrentTime) == false)
+                                    {
+                                        docsToArchive.Add((clonedId, null));
+                                        continue;
+                                    }
+                                    docsToArchive.Add((clonedId, document.Id));
+                                }
+                            } while (multiIt.MoveNext() && docsToArchive.Count + count < options.AmountToTake);
+                        }
+                    }
+
+                    count += docsToArchive.Count;
+                    if (docsToArchive.Count>0)
```

Comment:
Minor - formatting

Reply:
Fixed in 
https://github.com/ravendb/ravendb/pull/15715/commits/987c36644b842f1325ac27a52abbe473ec2b1a71